### PR TITLE
#524 eso module integration

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

Divide CICD into 2 part
CICD and ESO

Hide ESO options from dropdown on secret and secret override page
Show module installation link in the dropdown

Fixes # [(issue)](https://github.com/devtron-labs/dashboard/issues/524)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Verified hidden ESO options from dropdown on secret and secret override page in case of ESO module not installed
- Verified shown module installation link in the dropdown in case of ESO module not installed

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/secrets/secret.utils.tsx [hunk 1]
```diff
@@ -210,23 +213,37 @@ export const getTypeGroups = (typeValue?: string) => {
...
+                  options: esoGroups,
+              },
+              {
+                  label: 'Kubernetes External Secret (KES)',
+                  options: ksoGroups,
+              },
+          ]
+      }
+    } else {
+      if (typeValue) {
```
Comment (sohel-rp) [inline src/components/secrets/secret.utils.tsx:236]: Simply the else something like to reduce cyclomatic complexity,

```
return typeValue ? noGroups : [
              {
                  label: '',
                  options: noGroups,
              },
          ]
```

File: src/components/secrets/Secret.tsx [hunk 2]
```diff
@@ -973,13 +992,17 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
...
-                        value={codeEditorRadio === 'sample' ? sample : isESO ? esoSecretYaml: secretDataYaml}
+                        value={codeEditorRadio === 'sample' ? sample : isESO ? esoSecretYaml : secretDataYaml}
                         mode="yaml"
                         inline
                         height={350}
                         onChange={handleSecretDataYamlChange}
                         readOnly={secretMode && codeEditorRadio === 'sample'}
-                        shebang={codeEditorRadio === 'data' ? '#Check sample for usage.' : dataHeaders[externalType] || dataHeaders['default']}
+                        shebang={
+                            codeEditorRadio === 'data'
```
Comment (shivani170) [inline src/components/secrets/Secret.tsx:1002]: please use constant in place of `data`

File: src/components/EnvironmentOverride/SecretOverrides.tsx [hunk 3]
```diff
@@ -755,16 +766,18 @@ export function OverrideSecretForm({ name, appChartRef, toggleCollapse }) {
...
-                                onChange={changeEditorMode}
-                            >
-                                <RadioGroup.Radio value="gui">GUI</RadioGroup.Radio>
-                                <RadioGroup.Radio value="yaml">YAML</RadioGroup.Radio>
-                            </RadioGroup>}
+                            {!isESO && (
+                                <RadioGroup
+                                    className="gui-yaml-switch"
+                                    name="yaml-mode"
+                                    initialTab={yamlMode ? 'yaml' : 'gui'}
```
Comment (shivani170) [inline src/components/EnvironmentOverride/SecretOverrides.tsx:773]: Same here for yaml/gui

File: src/components/secrets/Secret.tsx [hunk 4]
```diff
@@ -973,13 +992,17 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
...
                         mode="yaml"
                         inline
                         height={350}
                         onChange={handleSecretDataYamlChange}
                         readOnly={secretMode && codeEditorRadio === 'sample'}
-                        shebang={codeEditorRadio === 'data' ? '#Check sample for usage.' : dataHeaders[externalType] || dataHeaders['default']}
+                        shebang={
+                            codeEditorRadio === CODE_EDITOR_RADIO_STATE.DATA
+                                ? '#Check sample for usage.'
+                                : dataHeaders[externalType] || dataHeaders['default']
```
Comment (shivani170) [inline src/components/secrets/Secret.tsx:1004]: Same for default

File: src/components/secrets/Secret.tsx [hunk 5]
```diff
@@ -909,16 +926,18 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
...
-                        onChange={changeEditorMode}
-                    >
-                        <RadioGroup.Radio value="gui">GUI</RadioGroup.Radio>
-                        <RadioGroup.Radio value="yaml">YAML</RadioGroup.Radio>
-                    </RadioGroup>}
+                    {!isESO && (
+                        <RadioGroup
+                            className="gui-yaml-switch"
+                            name="yaml-mode"
+                            initialTab={yamlMode ? 'yaml' : 'gui'}
```
Comment (shivani170) [inline src/components/secrets/Secret.tsx:933]: Same

File: src/components/secrets/Secret.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/secrets/Secret.tsx]: Done

File: src/components/secrets/Secret.tsx [hunk 6]
```diff
@@ -973,13 +992,17 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
             {(isHashiOrAWS || isESO) && yamlMode ? (
                 <div className="yaml-container">
                     <CodeEditor
-                        value={codeEditorRadio === 'sample' ? sample : isESO ? esoSecretYaml: secretDataYaml}
+                        value={codeEditorRadio === 'sample' ? sample : isESO ? esoSecretYaml : secretDataYaml}
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/secrets/Secret.tsx]: Done

File: src/components/secrets/Secret.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/secrets/Secret.tsx]: Done

File: src/components/secrets/Secret.tsx (hunk 5 above)
Comment (vivek-devtron) [reply src/components/secrets/Secret.tsx]: Done

**Review by shivani170 — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**